Code/Core/Network/TCPConnectionPool.cpp

```diff
@@ -456,7 +456,10 @@ bool TCPConnectionPool::SendInternal( const ConnectionInfo * connection, const T
     connection->m_InUse = true;
 #endif
 
```

Comment:
It should be impossible for  m_Socket to be INVALID_SOCKET while correctly using this class. If you saw the assert firing, it likely indicates a bug elsewhere in the code.

Reply:
Yes. There's probably a race condition in my code.
This assert is only happening on mac os.
How do I debug race conditions on Travis CI mac machines?
Adding printfs would probably change the timing...

I will try to reproduce it on a mac machine on my side.